modules/ROOT/pages/architecture/hexagonale_architecture.adoc

```diff
@@ -194,6 +194,180 @@ This makes the application more difficult to understand and harder to locate the
+
+With regard to the adapters, the recommendation is to create them within one module in the package structure mentioned above.
+This offers a good compromise between structuring and tailor-made dependencies and manageable complexity on the other hand.
+Given the usual level of complexity, it can be assumed that there are usually no more than 2-3 adapters, so structuring using packages is sufficient here.
+As complexity increases, two additional options become available:
+
+1. Separate modules for in and out. This allows the frameworks for incoming and outgoing adapters to be maintained separately and provides an additional level of structuring.
+2. Individual modules for each adapter. This means that each adapter can be viewed and developed completely isolated.
+The dependencies can also be maintained independently.
+
+With both options, the complexity increases accordingly.
+The recommendation is to start with a package structure and divide it further if necessary.
+
+The management of dependencies should aim to define them as far "down" as possible and not on the root or business component level.
+In particular, technical frameworks such as JPA should be defined directly in the adapter module so that they are not available in the core module.
+In order to keep versions consistent, dependency management in the master pom can be used in Maven.
+The advantage of this is that the version only needs to be defined at the main level and does not need to be specified in every adapter pom.
+This approach also prevents the same framework from being used in different versions in the adapters.
+
+It is necessary that the adapters can access the core and therefore have a dependency so that, on the one hand, incoming adapters can call logic in the core and, on the other hand, outgoing adapters from the core can be used.
+The latter is only possible via dependency inversion, as the core is not allowed to access the outgoing adapter directly.
+In this case, the out ports in the core are defined and used as an interface.
+The dependency injection framework in Spring or Quarkus then determines in the background the appropriate implementation from the out adapters that implement the out port interface.
+In this way, it is possible for the core to call functionality in the adapters without having a dependency in this direction.
+Under no circumstances should there be a dependency from the core to the adapter.
+This would mean that fundamental aspects of the hexagonal architecture could be undermined and technical frameworks could also be used in the core.
+
+NOTE: In case you are using Quarkus environment the dependency inversion is not possible out of the box.
+That means that outgoing ports cannot injected as expected which causes errors during application start.
+In that case you can inject outgoing ports with `Instance<YourOutgoingPort>` which decouples the injection process.
```

Comment:
Maybe a link to the issue or documentation describing this would be helpful in case it changes in the future.

Reply:
its not really an issue, its expected behavior: https://quarkus.io/guides/cdi#hm-wait-a-minute-what-happens-if-multiple-beans-declare-the-same-type

Will link it in the text.